src/routes/homework/CreateHomework.tsx

```diff
@@ -0,0 +1,492 @@
+      homeworkRequest.closed_questions > numberOfQuestions.closed
+    ) {
+      isValid = false;
+      toastDispatch(
+        updateToast({
+          title: 'Error',
+          type: 'error',
+          message: 'No has seleccionada suficientes preguntas cerradas',
+        })
+      );
+    }
+
+    return isValid;
+  };
+
+  const submitHomework = async (event: React.MouseEvent<HTMLButtonElement>) => {
+    event.preventDefault();
+
+    if (checkRules()) {
+      console.log('Submit!!');
+
+      inputErrorsDispatch({
+        type: 'reset',
+        payload: INITIAL_INPUT_ERRORS,
+      });
+
+      resetValues();
+
+      formRef.current?.reset();
+
+      toastDispatch(
+        updateToast({
+          title: 'Éxito',
+          type: 'success',
+          message: 'Se ha creado la tarea',
+        })
+      );
+
+      return;
+    }
+
+    inputErrorsDispatch({
+      type: 'update',
+      payload: { title: createHomeworkRules.title(homeworkRequest.title) },
+    });
+  };
+
+  const classElement = !id ? (
+    <AutocompleteField
+      key={autocompleteKey}
+      label="Clase"
+      id="class_id"
+      className={`${styles.input} ${styles.autocomplete}`}
+      error={inputErrors.class_id}
+      handleBlur={() => {}}
+      handleChange={autocompleteOnChangeHandler}
+      list={data as ItemList[]}
+    />
+  ) : (
+    <InputField
+      label="Clase"
+      type="text"
+      id="class_id"
+      required
+      className={`${styles.input} ${styles.autocomplete}`}
+      error=""
+      value=""
+      defaultValue={`[${(data as ItemList).id}] - ${(data as ItemList).value}`}
+      handleBlur={() => {}}
+      handleChange={() => {}}
+      readOnly
+    />
+  );
+
+  const questionNodes = homeworkRequest.questions_ids.map((question) => (
+    <li key={question.id}>
+      <QuestionCard
+        question={question}
+        onDelete={deleteQuestionNode}
+      />
+    </li>
+  ));
+
+  return (
+    <>
+      <SectionHeader
+        title={`Crear Tarea ${difficulty}`}
+        childType="backButton"
+      >
+        <Link
+          to="../.."
+          relative="path"
+          className={styles.icon}
+        >
+          <IconBack />
+        </Link>
+      </SectionHeader>
+      <div className={styles.container}>
+        <form
+          className={styles['form-container']}
+          ref={formRef}
+        >
+          <div className={styles['header-container']}>
+            <span className={styles.header}>Detalles</span>
+          </div>
+          <div>
+            <InputField
+              label="Título"
+              type="text"
+              id="title"
+              required
+              className={styles.input}
+              error={inputErrors.title}
+              value=""
+              handleBlur={() => {}}
+              handleChange={onInputChangeHandler}
+            />
+            {classElement}
+            <div className={styles['days-container']}>
+              <InputField
```

Comment:
Back in the join-list, I had a lot of InputFields, but I handle to make a map function to make this part shorter. 
I think this is a bit more complex, given all the different attributes, but I would think that would make this easier to read

Reply:
As discussed, I would prefer to keep the input JSX declaration as its easier for me to identify the input field props, than using a map.  We should definitely discuss this further.